.github/workflows/go.yml

```diff
@@ -0,0 +1,37 @@
+name: Go
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  autotests:
+    runs-on: ubuntu-latest
+    container: golang:1.17
+    
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Run autotests
+      run: |
+        go test -p 1 -race -vet=off -cover -coverprofile=coverage.out -covermode=atomic -v -coverpkg=./... -count=1 ./...
+        go tool cover -func=coverage.out
+    
+    - name: Upload coverage to Codecov
+      uses: codecov/codecov-action@v1
+      with:
+        token: ${{ secrets.CI_CODECOV_TOKEN }}
+        file: ./coverage.out
+        fail_ci_if_error: true
+
+  code-style:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+        
+    - name: Run golangci-lint
```

Comment:
Be sure to fix the linter version, otherwise CI is likely to be broken when updating it.